tests/cli/keyboard_interface.c

```diff
@@ -0,0 +1,163 @@
+        exit(0);
+    }
+
+    servaddr.sin_family = AF_INET;
+    servaddr.sin_addr.s_addr = htonl(INADDR_ANY);
+    servaddr.sin_port = htons(PORT);
+
+    if((bind(sockfd, (struct sockaddr*)&servaddr, sizeof(servaddr))) != 0){
+        printf("socket bind failed... %s\n", strerror(errno));
+        exit(0);
+    }
+
+    printf("Waiting for client\n");
+    if((listen(sockfd, 5)) != 0){
+        printf("listen failed...\n");
+        exit(0);
+    }
+    socklen_t len = sizeof(cli);
+    connfd = accept(sockfd, (struct sockaddr*)&cli, &len);
+    printf("Accepted client\n");
+    if (connfd < 0){
+        printf("server accept failed...\n");
+        exit(0);
+    }
+    return connfd;
+}
+
+int main() {
+    signal(SIGINT, sigint_handler);
+
+    // The bitmap of buttons pressed to be sent to net handler
+    uint32_t buttons = 0;
+    float joystick_vals[4] = {0};
+    // The received bitstring from Python to be parsed into BUTTONS
+    char buff[33];
+
+    printf("Its gamer time\n");
+
+    printf("Starting Net Handler CLI...\n");
+    fflush(stdout);
+
+    start_net_handler();
+    sleep(2);
+    // Connect dummy gamepad and start teleop mode
+    send_gamepad_state(0, joystick_vals);
+    send_run_mode(DAWN, TELEOP);
+    sleep(2);
+
+    // Start getting keyboard inputs
+    int fd = connect_tcp();
+
+    // Receive a bitstring of buttons and parse it
+    while(1) {
+        // Reset buttons/joysticks
+        buttons = 0;
+        memset(buff, 0, 32 * sizeof(char));
+        memset(joystick_vals, 0, 4 * sizeof(float));
+        // Receive bit string
+        recv(fd, buff, sizeof(buff), 0);
+        buff[32] = '\0';
+
+        // Parse joystick values
+        for(int i = joystick_left_x_right; i <= joystick_right_y_up; i++){
+            float pushed = 0;
+            if (buff[i] == '1'){
+                pushed = .25;
+            }
+            if (pushed != 0){
+                switch (i) {
+
+                    case 13:
+                        joystick_vals[0] = pushed;
+                        break;
+                    case 14:
+                        joystick_vals[0] = -1 * pushed;
+                        break;
+                    case 15:
+                        joystick_vals[1] = -1 * pushed;
+                        break;
+                    case 16:
+                        joystick_vals[1] = 1 * pushed;
+                        break;
+                    case 17:
+                        joystick_vals[2] = -1 * pushed;
+                        break;
+                    case 18:
+                        joystick_vals[2] = 1 * pushed;
+                        break;
+                    case 19:
+                        joystick_vals[3] = -1 * pushed;
+                        break;
+                    case 20:
+                        joystick_vals[3] = pushed;
+                        break;
+                }
+            }
+        }
+
+        // Parse the buttons TODO: EXTEND TO BUTTON_XBOX
```

Comment:
remove TODO, or fix the for loop because the xbox button is included right now

Reply:
@DanielMolina24 The loop is correct! You can remove the TODO

Reply:
Yea sorry forgot to remove it. Thx for reminding me